### ISSUE TITLE
Issues and suggestions at 3ID

### ISSUE THREAD

**Issue by hanfeiyan** [NSLS-II/Bug-Reports#4]
1-A position shows when a motor is moved in Ophyd using mov or movr command. Sometimes it shows the end position after the moving is completed (which it should do) but sometimes it doesn't.
2-Sometime the scan dies during a long scan. Couldn't even quit using Ctrl-c. Ken may already reported the problem.
3- When displaying the data, the first date and time column take too much space. It is not necessary to show the date for all the data points. Also, the data column should be aligned with the name. Sometimes it is very confusing if they are not aligned.
4- Please add a column to show the index of the data point so we know where we are during a long scan.
5-When using automatic update in replay, the whole program seems to be frozen and has extremely slow response to any operation. 
6-When displaying the data in replay, it is preferred to set the scanned motor as the default x axis, not the index.
7-Suggest to add FWHM, CEN, Peak position,... these statistics to the figure  
8-It's preferred to have error bar added to the plot

**Comment by danielballan** [comment]
Updates on your various comments:
1. This is probably still an issue. We have not addressed mov/movr, as this is somewhat separate from bluesky (no data is recorded).
2. This has been fixed.
3. This has been fixed.
4. This has been fixed.
5. Replay has been retired. Bluesky addresses live feedback; past scans can be viewed in a notebook or using the new "album" package, a web app that we are experimenting with in response to your feedback. We will demo this for you soon.
6. Bluesky does this. Album will do this. Replay is retired.
7. These are in the works, aiming to be deployed in the coming days.

**Comment by tacaswell** [comment]
For 8, where do the error values come from?

**Comment by hanfeiyan** [comment]
The most common one is the statistical error scaled with sqrt(n), where n is the counting number.

Hanfei

**Comment by hanfeiyan** [comment]
Hi guys,

Another two things I forget to mention. One is the scan ID, can we make the number unique? This is the only user trackable ID. We found before this number can be reset if the scan engine crashes or a remote session is used. This leads to two scans with the same ID number.

Second is that we need to add scan id into the long unique file name. There are two reasons why we want this. One is there are cases we need to access data directly. Second is that in case the metadata base crashes, we can still access the data.

Hanfei

**Comment by tacaswell** [comment]
Everything has a uid which is the unique id for that document.  You can search for scans based on the first 6 or so characters.

It is best to refer to what you call 'scan id' as 'transient id'.

**Comment by hanfeiyan** [comment]
I mean the scan number, which is often the only thing a user remember for a scan.

Hanfei

**Comment by danielballan** [comment]
If are you are worried about uniqueness, you should use the unique ID. That's just how it works.

**Comment by hanfeiyan** [comment]
The issue with unique id is that it is difficult to remember. It is good for machine, but not for human. 

Hanfei

**Comment by danielballan** [comment]
Well, more to your point, I think our new way of incrementing the scan ID happens to be more robust. I don't think a crash at any point in the scan could cause an ID to be duplicated; the new value is saved even before the scan is begun.

Give us a shout if you get a single example of this kind of failure in the next cycle.

**Comment by klauer** [comment]
@danielballan How does the `history` module (which is what is storing the 'transient' scan ID currently) handle multiple instantiations from different python sessions?

Here's a scenario: @hanfeiyan sometimes runs scans remotely. His scan finishes and someone at the beamline soon takes over. What's the next transient ID that's generated - is it aware that the remote session incremented the id?

**Comment by danielballan** [comment]
Between the call to the RunEngine and the opening of the new run, a SQLite
file records re incremented value. The next run will read that value, no
matter where it is run, as long as it has access to that file.

**Comment by danielballan** [comment]
@klauer Has HXN tried PeakStats yet?

**Comment by tacaswell** [comment]
There is still a race condition if two users try to run scans _at the same time_ but it that case you have bigger issues than the runs sharing a transient ID.

**Comment by klauer** [comment]
@danielballan Don't think so - will look into that

@tacaswell @danielballan admitted offline the `history` module doesn't work as mentioned above. I resorted to storing the beamline-global transient scan ID in EPICS, which is working well. That said, yes - there is still a race condition. I've tried to handle that at the user-level - they have a little textbox on the CSS screens they can use to communicate with others that intend to acquire data on the beamline. So during an experiment run locally, they may update it saying `hanfei at 3idc taking data` or something along those lines.  Additionally there's a CSS indicator showing that a bluesky scan is in progress, should they forget to update the message.

This is by no means perfect, but I think it's good enough.
